server/services/userService.test.ts

```diff
@@ -178,86 +99,72 @@ describe('User service', () => {
-      const defaultUser: User = {
-        ...defaultTokenData,
-        authSource: 'nomis',
-        token: TokenMock,
-        user_id: '12345',
-        roles: rolesForMockToken,
-      }
       it('returns cached data if number of caseloads is 1', async () => {
-        const cachedData: UserData = {
-          caseLoads: [{ caseloadFunction: '', caseLoadId: '1', currentlyActive: true, description: '', type: '' }],
-          activeCaseLoad: { caseloadFunction: '', caseLoadId: '1', currentlyActive: true, description: '', type: '' },
+        const cachedData: PrisonUserAccess = {
+          caseLoads: [{ caseloadFunction: '', caseLoadId: '123', currentlyActive: true, description: '', type: '' }],
+          activeCaseLoad: { caseloadFunction: '', caseLoadId: '123', currentlyActive: true, description: '', type: '' },
           services: [],
         }
-        const output = await userService.getUserData(defaultUser, cachedData)
-        expect(prisonApiClient.getUserCaseLoads).toBeCalledTimes(0)
-        expect(prisonApiClient.getUserLocations).toBeCalledTimes(0)
-        expect(prisonApiClient.getIsKeyworker).toBeCalledTimes(0)
+
+        cacheServiceMock.getData.mockResolvedValue(cachedData)
+
+        const output = await userService.getPrisonUserAccess(prisonUserMock)
+        expect(prisonApiClient.getUserCaseLoads).not.toHaveBeenCalled()
+        expect(prisonApiClient.getUserLocations).not.toHaveBeenCalled()
+        expect(prisonApiClient.getIsKeyworker).not.toHaveBeenCalled()
         expect(output).toEqual(cachedData)
       })
 
-      it('returns cached data if number of active caseload is unchanged', async () => {
-        const cachedData: UserData = {
+      it('returns cached data if active caseload is unchanged', async () => {
+        const cachedData: PrisonUserAccess = {
           caseLoads: [
             expectedCaseLoads[0],
-            { caseloadFunction: '', caseLoadId: '2', currentlyActive: true, description: '', type: '' },
+            { caseloadFunction: '', caseLoadId: '2', currentlyActive: false, description: '', type: '' },
           ],
           activeCaseLoad: expectedCaseLoads[0],
           services: [],
         }
-        const output = await userService.getUserData(defaultUser, cachedData)
+
+        cacheServiceMock.getData.mockResolvedValue(cachedData)
+
+        const output = await userService.getPrisonUserAccess(prisonUserMock)
         expect(prisonApiClient.getUserCaseLoads).toBeCalledTimes(1)
         expect(prisonApiClient.getUserLocations).toBeCalledTimes(0)
         expect(prisonApiClient.getIsKeyworker).toBeCalledTimes(0)
         expect(output).toEqual(cachedData)
       })
 
       it('gets new data if active caseload has changed', async () => {
-        const cachedData: UserData = {
+        const cachedData: PrisonUserAccess = {
           caseLoads: [
-            expectedCaseLoads[0],
+            { ...expectedCaseLoads[0], currentlyActive: false },
```

Comment:
Making the test a bit more realistic, only one of them would be `currentlyActive: true`